nanomq/pub_handler.c

```diff
@@ -1087,12 +1087,14 @@ handle_pub(nano_work *work, struct pipe_content *pipe_ct, uint8_t proto, bool is
 		return result;
 	}
 
+	topic = work->pub_packet->var_header.publish.topic_name.body;
+	uint32_t len = work->pub_packet->var_header.publish.topic_name.len;
 	// deal with topic alias
 	if (proto == MQTT_PROTOCOL_VERSION_v5) {
 		property_data *pdata = property_get_value(
 		    work->pub_packet->var_header.publish.properties,
 		    TOPIC_ALIAS);
-		if (work->pub_packet->var_header.publish.topic_name.len > 0) {
+		if (len > 0 && topic != NULL) {
 			if (pdata) {
 				dbhash_insert_atpair(work->pid.id,
 				    pdata->p_value.u16,
```

Comment:
Hey. I find a potential error. No length parameter of topic_name.body was passed to dbhash_insert_atpair function.

Reply:
decode_pub_message have added '\0' to the end of topic, thus we don't need parameter length here.

Reply:
Okay. Good!